### ISSUE TITLE
v5.3.1 Nested async error validation not working

### ISSUE THREAD

**Issue by philipchmalts** [erikras/redux-form#1516]
Hey,

Not sure if this is a regression or bug but I'm having a hard time doing async validation with dynamic fields. #400 and #429 say that it was fixed back in v4 but I am experiencing the same issue where asyncError ends up in the array level instead of the field level of nesting. 

Here's my asyncValidate function, note that the error object matches form. 

```
const asyncValidate = (values) =>
  new Promise((resolve, reject) => {
    if (!values.stocks[0].ticker) {
      reject({
        stocks: {
          0: {
            ticker: 'err'
          }
        }
      });
    } else {
      resolve();
    }
  });
```

However in the reducer the`asyncError` key ends up at the level of `0` instead of at `ticker`

Like this

![screen shot 2016-08-11 at 11 43 41 am](https://cloud.githubusercontent.com/assets/14876812/17594923/f36f38be-5fb8-11e6-82b1-ac6430663cb4.png)

Maybe I am misunderstanding how this is supposed to work, any help would be much appreciated!

**Comment by madisvain** [comment]
Having the same issue

**Comment by madisvain** [comment]
Ok - just upgraded everything to v6 and I can see the errors in the store but I think the format is wrong. @erikras could you give a hint to how to format the errors data to get them to propagate correctly?

<img width="243" alt="screenshot 2016-09-03 13 36 17" src="https://cloud.githubusercontent.com/assets/727994/18224370/7fcb53a4-71db-11e6-8992-5f111e04bf32.png">

**Comment by gordonk** [comment]
I'm up against this as well with an older 5.3.2 form.  
Validation errors on nested children (an array of objects) don't display.

A rejected promise of form...
`{ errors: { phone_numbers: [{description: 'broken0'}, { description: 'broken1' }]}}`

Results in a state tree looking like...

form
-> My Form
--> {other_form_fields)
--> phone_numbers
---> 0 
----> (other_phone_number_fields)
----> description
-----> touched: true
-----> _is_field_value: true
----> submitError
-----> description: "broken0"
---> 1
----> (other_phone_number_fields)
----> description
-----> touched: true
-----> _is_field_value: true
----> submitError
-----> description: "broken1"

**Comment by gordonk** [comment]
FWIW I've made some progress working out what's going wrong and I'll note my findings in case anyone wants to take it up.

The issue is in the ADD_ARRAY_VALUE function of the reducer.

When the value argument is undefined (which it is when an object without default values is added to the array) nothing gets populated in the child object EXCEPT a _isFieldValue: true attribute. Later, when errors are set or fields are focused/blurred in the form more attributes get added to the object, each of which is an object in its own right with a _isFieldValue attribute as well.

However, when the setErrors method iterates over the errors object and mashes them into the array of child objects, because there is already a _isFieldValue attribute alongside the child attributes so the submitError/AsyncError field gets written at that level rather than continuing down the tree into the foo attribute for example.

`const state = {
  last_name: {
    _isFieldValue: true
  },
  children: [
    {
      _isFieldValue: true,
      description: {
        _isFieldValue: true
      }
    }
  ],
}
console.log(state)

const errors = {
  last_name: "last name error",
  children: [
    {description: "description error"}
  ]
}
const newstate = {
  ...setErrors(state, errors, 'submitError')
};

console.log(newstate)
`

If this clarifies, with the  _isFieldValue: true attribute at the top level of the first child object in the arrray, the errors map incorrectly.  Remove that attribute and it correctly maps down into the final leaf of the description field.

The problem doesn't occur if something is provided in the forms initial state, even if its just a 
`children:{ foo: ""}` The downside of this, is that a pristine form will open with a nested child already populated, which may or may not be what you want.

Cause I was in a hurry and I'll be upgrading the library to 6 shortly, I chose to hack the reducer.js in the ADD_ARRAY_VALUE function. This might be useful to someone.

`    let newValue = makeFieldValue({value})
    if (value !== null && typeof value === 'object') {
      newValue = initializeState(value, fields || Object.keys(value))
    } else if (value === undefined && Array.isArray(fields)) {
      newValue = initializeState({}, fields)
    `
